### PR TITLE
Implement LFI

### PR DESCRIPTION
**Pull request by estringana**
### Description
First exploit prevention capability in PHP AppSec. This PR lays the foundation for the entire RASP architecture — not just LFI detection, but the full pipeline for reporting exploits, wiring into Remote Config, and emitting the right metrics.

Concretely:
- Wraps `file_get_contents`, `file_put_contents`, `fopen`, and `readfile` to
  push addresses to the WAF on each call
- Introduces exploit prevention metrics and the RASP span tags
  (`rasp.duration`, `rasp.duration_ext`)
- Adds the LFI capability to Remote Config so it can be toggled without a redeploy
- Adds `DD_APPSEC_RASP_ENABLED` config flag (disabled by default)
- Sets up the integration in a way that makes adding new attack types straightforward — new capabilities plug into the same architecture without touching the lower layers

Related Jiras: [APPSEC-52929], [APPSEC-53812], [APPSEC-53813]

[APPSEC-52929]: https://datadoghq.atlassian.net/browse/APPSEC-52929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APPSEC-53812]: https://datadoghq.atlassian.net/browse/APPSEC-53812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APPSEC-53813]: https://datadoghq.atlassian.net/browse/APPSEC-53813?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cataphract — CHANGES_REQUESTED** [review]

File: appsec/src/extension/configuration.h [hunk 1]
```diff
@@ -47,6 +47,7 @@ extern bool runtime_config_first_init;
     SYSCFG(BOOL, DD_APPSEC_HELPER_LAUNCH, "true")                                                                                     \
     CONFIG(STRING, DD_APPSEC_HELPER_PATH, DD_BASE("bin/libddappsec-helper.so"))                                                       \
     SYSCFG(BOOL, DD_APPSEC_STACK_TRACE_ENABLED, "true")                                                                               \
+    SYSCFG(BOOL, DD_APPSEC_RASP_ENABLED , "false")                                                                                    \
```
[the remark this answers is not in the document]
Comment (Anilm3) [reply appsec/src/extension/configuration.h:50]: Can we not just disable the RASP-specific integrations when this setting is `false`? That's what every other library does.

File: src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private static function preHook($variant)
+    {
+        return static function (HookData $hook) use ($variant) {
+            if (count($hook->args) == 0 || !is_string($hook->args[0])) {
+                return;
+            }
+
+            \datadog\appsec\push_address("server.io.fs.file", $hook->args[0], true);
```
Comment (cataphract) [inline src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php:66]: Isn't lfi only for local files? Then this should look at local files.

File: appsec/src/extension/configuration.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (cataphract) [reply appsec/src/extension/configuration.h:50]: Should be possible, you just need to untangle all the macro magic in integrations C. Right now it decides whether an integration is enabled like this:

```
        .is_enabled = get_DD_TRACE_##id##_ENABLED,              \
```

so we need to replace that pointer to require that DD_APPSEC_RASP_ENABLED be true.

File: appsec/src/extension/ddappsec.c [hunk 3]
```diff
@@ -541,6 +562,17 @@ static PHP_FUNCTION(datadog_appsec_push_address)
             dd_request_abort_redirect();
         }
     }
+
+    if (rasp && elapsed == 0) {
```
[the remark this answers is not in the document]
Comment (estringana) [reply appsec/src/extension/ddappsec.c:566]: Sorry but I didn't get this. Can you rephrase plz?

File: appsec/src/extension/configuration.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (estringana) [reply appsec/src/extension/configuration.h:50]: I went for another approach. Checking the env variable before wrappping the functions

```
class FilesystemIntegration extends Integration
{
    const NAME = "filesystem";

    public function init(): int
    {
        if (\dd_trace_env_config("DD_APPSEC_RASP_ENABLED")) {
            return Integration::LOADED;
        }
...
```

File: src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php (hunk 2 above)
Comment (estringana) [reply src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php:79]: Filtered

File: src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+{
+    const NAME = "filesystem";
+
+    public function init(): int
+    {
+        if (!\dd_trace_env_config("DD_APPSEC_RASP_ENABLED")) {
+            return Integration::LOADED;
+        }
+
+        if (!function_exists('\datadog\appsec\push_address')) {
```
Comment (bwoebi) [inline src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php:19]:
```suggestion
        if (!function_exists('datadog\appsec\push_address')) {
```
Nit: the FQN is really without leading slash. Just some places in PHP allow it and strip it away...

File: appsec/src/extension/ddappsec.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (cataphract) [reply appsec/src/extension/ddappsec.c:568]: You have two places where you calculate the elapsed time and call `dd_tags_add_rasp_duration_ext`. The first suffices if you remove `(res == dd_should_block || res == dd_should_redirect)` as part of the condition

File: appsec/src/extension/configuration.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (cataphract) [reply appsec/src/extension/configuration.h:50]: The disadvantage of this is that telemetry still reports this integration has enabled. Check with @Anilm3 if he's fine with that.

[the remark this answers is not in the document]
Comment (Anilm3) [reply appsec/src/extension/configuration.h:50]: I haven't really checked, but if you returned `Integration::NOT_LOADED` would telemetry still report it as enabled?

[the remark this answers is not in the document]
Comment (Anilm3) [reply appsec/src/extension/configuration.h:50]: How was this resolved, @estringana?

File: src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            'stat',
+            self::preHook('stat'),
+            null
+        );
+
+        \DDTrace\install_hook(
+            'lstat',
+            self::preHook('lstat'),
+            null
+        );
```
Comment (Anilm3) [inline src/DDTrace/Integrations/Filesystem/FilesystemIntegration.php:58]: We can skip `stat` and `lstat` as I don't think these would be relevant targets for LFI.

**Review by Leiyks — APPROVED** [review]

File: tests/ext/telemetry/integration_filesystem.phpt [hunk 6]
```diff
@@ -0,0 +1,105 @@
+--TEST--
+Filesystem integration depends on RASP. If RASP enabled, integration is enabled
```
[the remark this answers is not in the document]
Comment (estringana) [reply tests/ext/telemetry/integration_filesystem.phpt]: You're right. Sorted

**Review by cataphract — APPROVED** [review]

**Pull request merged**